### ISSUE TITLE
Incomplete downcast implementation for IError in `or` blocks

### ISSUE THREAD

**Issue by knz** [vlang/v#13296]
V full version: V 0.2.4 daa5be4.9dce819

**tldr**: given an interface that embeds `IError`, casting using `as` in an `or` block works but not `is` (nor `match`). This  does not happen with other interface combinations.

**What did you do?**

I am trying to use a family of custom error types that all implement `IError`, and share some other interface besides `IError`.

The program starts as follows, and this compiles fine:

```v
// Common interface to all error custom types.
interface ESpeaker {
        IError
        speak() string
}

// One custom error implementation.
struct MyError {
        // Mandatory fields from IError.
        msg  string
        code int
        // Custom field.
        blah string
}

// Interface implementation for this example custom type.
fn (e MyError) speak() string {
        return e.blah
}

// An example function that returns a custom error.
fn foo() ?string {
        return IError(MyError{
                msg: 'foo'
                blah: 'world'
        })
}
```

So far, so good. 

How does this get used? If we assume that _all_ errors in the program are always implementing `ESpeaker`, we can do this:
```v
fn main() {
        x := foo() or {
                s := err as ESpeaker
                s.speak()
        }
        println(x)
}
```
This works for the code above.  

However, what if some errors do not implement ESpeaker? 

In this case, we'd like to write this:

```v
fn main() {
        x := foo() or {
                if err is Speaker { err.speak() } else { 'undefined' }
        }
        println(x)
}
```
However, this **fails to compile** with:
```
test.v:25:27: error: unknown method or field: `IError.speak`.
1 possibility: `str`.
   23 | fn main() {
   24 |     x := foo() or {
   25 |         if err is Speaker { err.speak() } else { 'undefined' }
      |                                 ~~~~~~~
   26 |     }
   27 |     println(x)
```

Ditto, using `match`:
```v
fn main() {
        x := foo() or {
                match err {
                        ESpeaker { err.speak() }
                        else { 'undefined ' }
                }
        }
```
This fails with:
```
test.v:26:4: error: cannot implement interface `IError` with a different interface `ESpeaker`
   24 |     x := foo() or {
   25 |         match mut err {
   26 |             Speaker {
      |             ~~~~~~~
```

**What did you expect to see?**

I'm expecting that if `as` can cast, then `as` and `match` should be able to test the cast as well.
